squash_bot/match_tracker/orderers.py

```diff
@@ -0,0 +1,9 @@
+import abc
+
+from squash_bot.match_tracker.data import dataclasses
+
+
+class Orderer(abc.ABC):
+    @classmethod
```

Comment:
(Question) what does `@classmethod` add that `@abc.abstractmethod` doesn't already imply?

Reply:
They are two different things, `@classmethod` marks the method as a class method (a method that is called on the class itself rather than an instance of the class, allowing `Orderer.order()` rather than `Orderer().order()`)
`abstractmethod` comes from `abc` and throws an error if you try to call this method without overriding it

Reply:
Ah ok so that would be the equivalent of a static method then?

Reply:
No, static methods are different again, they change the signature of a method so they don't get a reference to the class itself

```py
class SomeClass:
  def some_method(self):
      ...
```

```py
class SomeClass:
  @classmethod
  def some_class_method(cls):
      ...
```

```py
class SomeClass:
  @staticmethod
  def some_static_method():
      ...
```